src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -395,15 +417,15 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
         }
 
         // process the base list for one part of a partial class, or for the only part of any other type declaration.
-        private Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> MakeOneDeclaredBases(ConsList<TypeSymbol> newBasesBeingResolved, SingleTypeDeclaration decl, BindingDiagnosticBag diagnostics)
+        private (TypeWithAnnotations localBase, ImmutableArray<NamedTypeSymbol> interfaces) MakeOneDeclaredBases(ConsList<TypeSymbol> newBasesBeingResolved, SingleTypeDeclaration decl, BindingDiagnosticBag diagnostics)
```

Comment:
>TypeWithAnnotations [](http://example.com/codeflow?start=17&length=19)

It feels like the change from NamedTypeSymbol to TypeWithAnnotations is not appropriate here because it looks like nullable annotations are not allowed at the top level of a base type, they are meaningless there and we don't need to look at them..

Reply:
Consider reverting all changes in this function, perhaps with exception of the change from Tuple to ValueTuple.